xcbgen-rs/src/defs.rs

```diff
@@ -670,23 +1097,30 @@ impl Alignment {
 
     /// Returns whether `self` meets the alignment requirements
     /// of `required`.
+    // FIXME: Does something that is aligned at a 5 byte boundary really meet a 4 byte alignment? I
+    // think `self.align >= required.align` really should be `self.align % required.align == 0`.
+    // Perhaps some unit tests for this?
```

Comment:
For now, `align` must be a power of 2. Maybe this could be generalized to any number?

Reply:
Generalizing alignment to non-power-of-2 might require calculating MCDs at some point.

Reply:
In that case I would like to make it not possible to create other instances of this. Like, having a non-`pub` field and having a `new` method that checks this kind of invariant.

(MCD? From context, I would guess "[something] common denominator", but I only know about least or greatest one. Either one is not too hard to compute. If you want, I can write something up.)

Oh and: Once again thanks a lot for writing all this. It's amazing. And sorry that I get lost at some parts of it.

Reply:
oops, I wanted to say GCD

I have created https://github.com/psychon/x11rb/pull/397 to make fields private and enforce alignments to be a power of two.